rust/src/dcerpc/dcerpc_udp.rs

```diff
@@ -67,37 +69,48 @@ pub struct DCERPCUDPState {
 impl DCERPCUDPState {
     pub fn new() -> DCERPCUDPState {
         return DCERPCUDPState {
+            tx_id: 0,
             header: None,
-            request: None,
-            response: None,
+            transactions: Vec::new(),
             fraglenleft: 0,
             uuid_entry: None,
             uuid_list: Vec::new(),
             de_state: None,
         };
     }
 
-    fn new_request(&mut self) {
-        let request = DCERPCRequest::new();
-        self.request = Some(request);
+    fn create_tx(&mut self, serial_no: u16) -> DCERPCTransaction {
+        let mut tx = DCERPCTransaction::new();
+        let endianness = self.get_hdr_drep_0() & 0x10;
+        tx.id = self.tx_id;
+        tx.call_id = serial_no as u32;
+        tx.endianness = endianness;
+        self.tx_id += 1;
+        tx
     }
 
-    fn new_response(&mut self) {
-        let response = DCERPCResponse::new();
-        self.response = Some(response);
+
+    fn evaluate_serial_no(&mut self) -> u16 {
+        let mut serial_hi: u8 = 0;
+        let mut serial_lo: u8 = 0;
+        if let Some(ref hdr) = &self.header {
+            serial_hi = hdr.serial_hi;
+            serial_lo = hdr.serial_lo;
+        }
+        let mut serial_no = Cursor::new(vec![serial_hi, serial_lo]);
```

Comment:
What is a Cursor and what does it do here?

Reply:
Cursor is used to allow reads or writes on in-memory buffers from what I understand. I picked it up from the example usage of [byteorder](https://docs.rs/byteorder/1.1.0/byteorder/trait.ReadBytesExt.html#method.read_u16) because I wanted to ensure that the data is read according to endianness.
But, there is a mistake. I have hard coded `LittleEndian` here so doing this all was useless. Changing it with the endianness received from header might be the way to go about it.

Do you think the same? Am I doing something wrong?